pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -232,10 +233,43 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 			portable:         volume.Portable,
 			crushDeviceClass: volume.CrushDeviceClass,
 			schedulerName:    volume.SchedulerName,
+			encrypted:        volume.Encrypted,
 		}
 
 		logger.Debugf("osdProps are %+v", osdProps)
 
+		// If the deviceSet template has "encrypted" but the Ceph version is not compatible
+		if osdProps.encrypted && !c.clusterInfo.CephVersion.IsAtLeast(cephVolumeRawEncryptionModeMinCephVersion) {
```

Comment:
Don't we also need to check for 15.2.5 if they are using octopus?

Reply:
Isn't 15.2.5 at least 14.2.11? 🙂

Reply:
Yes, but 15.2.0-15.2.4 are not valid, right?

Reply:
Oh yes, will add it in https://github.com/rook/rook/pull/5977, thanks!

Reply:
https://github.com/rook/rook/pull/5977/commits/fc35f3b9c1940f9c75c91f58ba58fe6f8ba7b07f